docs/source/rpc.rst

```diff
@@ -122,8 +116,82 @@ how a given function should be treated on the callee side.
 
 .. autofunction:: torch.distributed.rpc.functions.async_execution
 
-.. _rref:
 
+.. _rpc-backends:
+
+RPC Backends
+------------
+
+The RPC module can leverage different backends to perform the communication
+between the nodes. The backend to be used can be specified in the
+:func:`~torch.distributed.rpc.init_rpc` function, by passing a certain value of
+the :class:`~torch.distributed.rpc.BackendType` enum. Regardless of what backend
+is used, the rest of the RPC API won't change. Each backend also defines its own
+subclass of the :class:`~torch.distributed.rpc.RpcBackendOptions` class, an
+instance of which can also be passed to :func:`~torch.distributed.rpc.init_rpc`
+to configure the backend's behavior.
+
+.. autoclass:: BackendType
+
+.. autoclass:: RpcBackendOptions
+    :members:
+
+
+Process Group Backend
+^^^^^^^^^^^^^^^^^^
+
+The Process Group agent, which is the default, instantiates a process group from
+the :mod:`~torch.distributed` module and utilizes its point-to-point
+communication capabilities to send RPC messages across. Internally, the process
+group uses `the Gloo library <https://github.com/facebookincubator/gloo/>`_.
+
+Gloo has been hardened by years of extensive use in PyTorch and is thus very
+reliable. However, as it was designed to perform collective communication, it
+may not always be the best fit for RPC. For example, each networking operation
+is synchronous and blocking, which means that it cannot be run in parallel with
+others. Moreover, it opens a connection between all pairs of nodes, and brings
+down all of them when one fails, thus reducing the resiliency and the elasticity
+of the system. These limitations, and others, might have been mitigated but,
+instead, a separate backend, based on a more tailored library, was introduced to
+address them more fundamentally.
+
+.. autoclass:: ProcessGroupRpcBackendOptions
+    :members:
+    :inherited-members:
+
+
+TensorPipe Backend
+^^^^^^^^^^^^^^^^^^
+
+The TensorPipe agent leverages `the TensorPipe
+library <https://github.com/pytorch/tensorpipe>`_, which provides a natively
```

Comment:
natively provides or just native?

Reply:
I meant it as "it is point-to-point in a native way", rather than being collective with point-to-point being an "abuse" of the API.